horizon/facts/client.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional, Annotated
+from urllib.parse import urljoin
+
+from fastapi import HTTPException, Depends
+from httpx import AsyncClient, Request as HttpxRequest, Response as HttpxResponse
+from loguru import logger
+from starlette import status
+from starlette.requests import Request as FastApiRequest
+from starlette.responses import Response as FastApiResponse, StreamingResponse
+
+from config import sidecar_config
+from startup.remote_config import get_remote_config
+
+
+class FactsClient:
+    def __init__(self):
+        self._client: Optional[AsyncClient] = None
+
+    @property
+    def client(self) -> AsyncClient:
+        if self._client is None:
+            self._client = AsyncClient(
+                base_url=sidecar_config.CONTROL_PLANE,
+                headers={"Authorization": f"Bearer {sidecar_config.API_KEY}"},
+            )
+        return self._client
+
+    async def build_forward_request(
+        self, request: FastApiRequest, path: str
+    ) -> HttpxRequest:
+        """
+        Build an HTTPX request from a FastAPI request to forward to the facts service.
+        :param request: FastAPI request
+        :param path: Backend facts service path to forward to
+        :return: HTTPX request
+        """
+        forward_headers = {
+            key: value
+            for key, value in request.headers.items()
+            if key.lower() in {"authorization", "content-type"}
+        }
+        remote_config = get_remote_config()
```

Comment:
the reason is because he needed the remote config and propagating it all the way was really hard. 
there was only one place that called it so he created a singleton for that